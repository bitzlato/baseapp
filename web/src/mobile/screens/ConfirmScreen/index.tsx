import * as React from 'react';
import { useSelector } from 'react-redux';
import { useHistory } from 'react-router';
import { BackButtonMobile } from 'web/src/components/shared/Header/BackButtonMobile';
import { useT } from 'web/src/hooks/useT';
import { getVerificationStep } from '../../../helpers';
import { selectLabelData } from '../../../modules/user/kyc/label';
import { ConfirmScreen } from '../../../screens/ConfirmScreen';
import { MobileModal } from '../../components/Modal';

const ConfirmMobileScreen: React.FC = () => {
  const t = useT();
  const history = useHistory();
  const labels = useSelector(selectLabelData);
  const step = getVerificationStep(labels);

  if (step === 'phone') {
    return (
      <>
        <BackButtonMobile to="/">{t('page.mobile.confirm.back')}</BackButtonMobile>
        <div className="cr-mobile-confirm">
          <div className="cr-mobile-confirm__phone">
            <MobileModal
              isOpen
              onClose={() => history.goBack()}
              title={t('page.body.profile.header.account.profile.phone.unverified.title')}
            >
              <ConfirmScreen />
            </MobileModal>
          </div>
        </div>
      </>
    );
  }

  return (
    <div className="cr-mobile-confirm">
      <div className="cr-mobile-confirm__identity">
        <ConfirmScreen />
      </div>
    </div>
  );
};

export { ConfirmMobileScreen };
