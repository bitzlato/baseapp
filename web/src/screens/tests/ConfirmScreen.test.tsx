import { shallow } from 'enzyme';
import { TestComponentWrapper } from 'src/lib/test';
import { ConfirmScreen } from '../ConfirmScreen';

const setup = () =>
  shallow(
    <TestComponentWrapper>
      <ConfirmScreen />
    </TestComponentWrapper>,
  );

describe('ConfirmScreen test', () => {
  it('should render', () => {
    const wrapper = setup().render();
    expect(wrapper).toMatchSnapshot();
  });
});
