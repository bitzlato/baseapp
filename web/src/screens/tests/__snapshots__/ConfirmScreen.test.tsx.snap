// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmScreen test should render 1`] = `
<div
  class="pg-container pg-confirm"
>
  <div
    class="pg-confirm__logo"
  >
    <div
      class="pg-logo logo"
    >
      <svg
        class="pg-logo__img"
        fill="none"
        height="20"
        style="width:auto"
        viewBox="0 0 118 20"
        width="118"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0.384613 17.1605C0.384613 18.7911 1.19766 19.5924 2.8651 19.5924H8.21193C12.1945 19.5924 14.7577 17.4698 14.7577 14.1804C14.7577 11.5658 13.104 9.7384 10.5684 9.49943V9.33075C12.5803 8.85281 13.793 7.29248 13.793 5.1277C13.793 2.23196 11.6433 0.404545 8.22571 0.404545H2.8651C1.19766 0.404545 0.384613 1.21985 0.384613 2.83641V17.1605ZM4.07778 8.38893V3.25812H7.37132C9.10766 3.25812 10.1274 4.21399 10.1274 5.77432C10.1274 7.44711 9.02498 8.38893 6.53071 8.38893H4.07778ZM4.07778 16.7388V11.0316H7.60559C9.78291 11.0316 10.968 12.1421 10.968 13.9555C10.968 15.7829 9.76912 16.7388 6.83388 16.7388H4.07778Z"
          fill="var(--primary-text-color)"
        />
        <path
          d="M18.0925 19.761C19.0847 19.761 19.636 19.2409 20.0218 17.9196L21.0691 14.6584H27.67L28.7311 17.9196C29.1031 19.2269 29.6819 19.761 30.6879 19.761C31.749 19.761 32.4518 19.1144 32.4518 18.1023C32.4518 17.7087 32.3553 17.2448 32.1624 16.6404L27.4219 2.78018C26.8294 0.980882 26.0025 0.249918 24.4453 0.249918C22.8744 0.249918 22.0338 0.994939 21.4412 2.78018L16.7007 16.6404C16.4664 17.3573 16.37 17.7931 16.37 18.1726C16.37 19.1285 17.0728 19.761 18.0925 19.761ZM21.896 11.8329L24.2938 4.07342H24.4729L26.8569 11.8329H21.896Z"
          fill="var(--primary-text-color)"
        />
        <path
          d="M34.2019 15.6423C34.2019 16.4436 34.6705 17.3011 35.5386 18.1164C36.7789 19.3112 38.9424 20 41.3816 20C45.874 20 48.7266 17.6806 48.7266 14.0258C48.7266 11.2003 47.1005 9.49943 43.6553 8.69818L40.8303 8.00939C39.0802 7.60173 38.2258 6.75831 38.2258 5.56347C38.2258 4.07342 39.4799 3.11755 41.4367 3.11755C43.0628 3.11755 44.3306 3.83446 45.1712 5.11364C45.6122 5.67592 46.1083 5.92895 46.6871 5.92895C47.5828 5.92895 48.1891 5.3245 48.1891 4.43891C48.1891 3.55331 47.7344 2.65367 46.88 1.86647C45.6397 0.699742 43.6278 0.0390625 41.3816 0.0390625C37.1923 0.0390625 34.4638 2.33035 34.4638 5.85866C34.4638 8.64195 36.0761 10.3991 39.3145 11.1582L42.1257 11.861C44.1652 12.353 45.0472 13.1543 45.0472 14.4475C45.0472 16.0078 43.7932 16.9075 41.5883 16.9075C39.9484 16.9075 38.5428 16.2609 37.4817 15.0098C36.8202 14.2788 36.3103 14.0117 35.6902 14.0117C34.8496 14.0117 34.2019 14.6865 34.2019 15.6423Z"
          fill="var(--primary-text-color)"
        />
        <path
          d="M51.7583 17.1605C51.7583 18.7911 52.5713 19.5924 54.2388 19.5924H62.5208C63.4992 19.5924 64.1332 18.9879 64.1332 18.0461C64.1332 17.1043 63.4992 16.4858 62.5208 16.4858H55.4514V11.3409H62.1763C63.0583 11.3409 63.6508 10.7786 63.6508 9.89303C63.6508 9.00743 63.0583 8.4311 62.1763 8.4311H55.4514V3.51114H62.5622C63.4992 3.51114 64.1332 2.89264 64.1332 1.96487C64.1332 1.03711 63.4992 0.404545 62.5622 0.404545H54.2388C52.5713 0.404545 51.7583 1.21985 51.7583 2.83641V17.1605Z"
          fill="var(--primary-text-color)"
        />
        <path
          clip-rule="evenodd"
          d="M75.0182 0C73.5028 0 72.2744 1.25311 72.2744 2.79891V17.1933C72.2744 18.7391 73.5028 19.9922 75.0182 19.9922H114.314C115.829 19.9922 117.058 18.7391 117.058 17.1933V2.79891C117.058 1.25311 115.829 0 114.314 0H75.0182ZM84.8718 14.3366H84.7998C84.1516 15.407 83.0509 15.9841 81.6826 15.9841C79.5737 15.9841 78.154 14.6199 78.154 12.5946C78.154 10.5902 79.584 9.34147 81.9604 9.34147H84.8203V8.27109C84.8203 6.8544 83.987 6.13032 82.4336 6.13032C81.518 6.13032 80.9008 6.45563 80.2527 7.2112C80.0058 7.50503 79.7692 7.60997 79.4503 7.60997C79.0388 7.60997 78.761 7.32663 78.761 6.89638C78.761 6.27724 79.1828 5.66859 80.0469 5.16488C80.6745 4.79759 81.4872 4.5982 82.5468 4.5982C85.1495 4.5982 86.5898 5.89945 86.5898 8.26059V14.9138C86.5898 15.5434 86.2709 15.9107 85.7256 15.9107C85.201 15.9107 84.8923 15.5539 84.8718 14.9557V14.3366ZM79.9646 12.5526C79.9646 11.4298 80.7259 10.8001 82.053 10.8001H84.8203V12.0489C84.8203 13.4236 83.6578 14.473 82.1662 14.473C80.8494 14.473 79.9646 13.6965 79.9646 12.5526ZM91.4146 18.828V14.1792H91.5072C92.001 15.3335 93.1224 15.9841 94.6141 15.9841C97.1962 15.9841 98.8216 14.1687 98.8216 11.2619V9.30999C98.8216 6.42415 97.2168 4.5982 94.6758 4.5982C93.1224 4.5982 91.8673 5.33278 91.4249 6.52909H91.3323V5.65809C91.3323 5.02846 91.0237 4.67166 90.4888 4.67166C89.9538 4.67166 89.6452 5.03895 89.6452 5.67908V18.828C89.6452 19.4576 89.9744 19.8354 90.5299 19.8354C91.0854 19.8354 91.4146 19.4576 91.4146 18.828ZM91.4146 9.40443V11.3773C91.4146 13.1718 92.5874 14.41 94.2849 14.41C96.0337 14.41 97.0419 13.1193 97.0419 10.8841V9.68777C97.0419 7.44206 96.0337 6.1723 94.2334 6.1723C92.4948 6.1723 91.4146 7.38959 91.4146 9.40443ZM103.348 14.1792V18.828C103.348 19.4576 103.019 19.8354 102.463 19.8354C101.908 19.8354 101.579 19.4576 101.579 18.828V5.67908C101.579 5.03895 101.887 4.67166 102.422 4.67166C102.957 4.67166 103.266 5.02846 103.266 5.65809V6.52909H103.358C103.801 5.33278 105.056 4.5982 106.609 4.5982C109.15 4.5982 110.755 6.42415 110.755 9.30999V11.2619C110.755 14.1687 109.13 15.9841 106.548 15.9841C105.056 15.9841 103.935 15.3335 103.441 14.1792H103.348ZM103.348 11.3773V9.40443C103.348 7.38959 104.428 6.1723 106.167 6.1723C107.967 6.1723 108.975 7.44206 108.975 9.68777V10.8841C108.975 13.1193 107.967 14.41 106.218 14.41C104.521 14.41 103.348 13.1718 103.348 11.3773Z"
          fill="var(--icons)"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="beta"
      >
        <svg
          viewBox="0 0 229.6 70.801"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 36.2 70.001 L 0 70.001 L 0 0.001 L 34.2 0.001 Q 47 0.001 53.85 4.901 A 16.615 16.615 0 0 1 58.533 9.858 Q 60.7 13.476 60.7 18.201 A 18.125 18.125 0 0 1 59.836 23.869 A 16.179 16.179 0 0 1 58.25 27.301 A 17.095 17.095 0 0 1 51.673 33.455 A 20.276 20.276 0 0 1 51.4 33.601 A 20.655 20.655 0 0 1 56.676 36.089 A 17.126 17.126 0 0 1 60.8 39.951 Q 64.2 44.401 64.2 50.901 A 18.849 18.849 0 0 1 63.195 57.203 A 15.81 15.81 0 0 1 57.05 65.051 Q 49.9 70.001 36.2 70.001 Z M 229.6 39.301 L 229.6 70.001 L 215 70.001 L 215 63.301 A 14.124 14.124 0 0 1 206.886 69.633 Q 204.333 70.472 201.17 70.708 A 34.561 34.561 0 0 1 198.6 70.801 A 30.895 30.895 0 0 1 193.491 70.399 Q 190.506 69.898 188.01 68.774 A 19.399 19.399 0 0 1 187.85 68.701 Q 183.3 66.601 180.9 62.901 Q 178.5 59.201 178.5 54.501 A 15.95 15.95 0 0 1 179.415 48.984 A 13.741 13.741 0 0 1 184.15 42.701 Q 189.575 38.572 200.671 38.407 A 62.772 62.772 0 0 1 201.6 38.401 L 214 38.401 Q 214 33.301 210.9 30.551 Q 208.536 28.453 204.368 27.956 A 23.379 23.379 0 0 0 201.6 27.801 Q 197.3 27.801 193.15 29.151 A 25.36 25.36 0 0 0 189.203 30.795 A 19.737 19.737 0 0 0 186.1 32.801 L 180.5 21.901 A 30.462 30.462 0 0 1 186.342 18.713 A 39.69 39.69 0 0 1 191.05 17.101 Q 197.2 15.401 203.7 15.401 Q 216.2 15.401 222.9 21.351 A 18.77 18.77 0 0 1 228.363 30.105 Q 229.6 34.144 229.6 39.301 Z M 127.8 47.501 L 87.1 47.501 Q 88.2 52.501 92.3 55.401 A 15.411 15.411 0 0 0 98.206 57.893 A 21.539 21.539 0 0 0 102.5 58.301 A 24.066 24.066 0 0 0 106.346 58.007 A 18.051 18.051 0 0 0 109.95 57.051 Q 113.2 55.801 116 53.101 L 124.3 62.101 Q 116.867 70.609 102.738 70.796 A 48.235 48.235 0 0 1 102.1 70.801 Q 93 70.801 86 67.251 Q 79 63.701 75.2 57.401 A 26.625 26.625 0 0 1 71.46 45.089 A 32.401 32.401 0 0 1 71.4 43.101 A 28.604 28.604 0 0 1 73.015 33.399 A 26.419 26.419 0 0 1 75.15 28.851 Q 78.9 22.501 85.45 18.951 A 29.628 29.628 0 0 1 97.853 15.471 A 35.551 35.551 0 0 1 100.1 15.401 A 31.66 31.66 0 0 1 109.389 16.729 A 28.204 28.204 0 0 1 114.4 18.801 Q 120.8 22.201 124.45 28.551 A 27.345 27.345 0 0 1 127.803 38.677 A 35.01 35.01 0 0 1 128.1 43.301 A 3.074 3.074 0 0 1 128.095 43.432 Q 128.065 44.046 127.833 47.071 A 1232.566 1232.566 0 0 1 127.8 47.501 Z M 169 56.401 L 173.2 67.401 A 14.169 14.169 0 0 1 170.28 69.044 A 19.021 19.021 0 0 1 167.55 69.951 A 28.185 28.185 0 0 1 161.42 70.787 A 31.655 31.655 0 0 1 160.5 70.801 A 30.235 30.235 0 0 1 154.668 70.273 Q 151.236 69.598 148.572 68.059 A 15.975 15.975 0 0 1 145.65 65.901 Q 140.4 61.001 140.4 51.501 L 140.4 29.401 L 132.1 29.401 L 132.1 17.401 L 140.4 17.401 L 140.4 4.301 L 156 4.301 L 156 17.401 L 169.4 17.401 L 169.4 29.401 L 156 29.401 L 156 51.301 A 10.606 10.606 0 0 0 156.188 53.365 Q 156.43 54.579 156.979 55.52 A 5.618 5.618 0 0 0 157.75 56.551 A 5.696 5.696 0 0 0 160.587 58.162 Q 161.49 58.386 162.557 58.4 A 10.904 10.904 0 0 0 162.7 58.401 A 11.385 11.385 0 0 0 165.921 57.963 A 9.306 9.306 0 0 0 169 56.401 Z M 35 40.501 L 16.1 40.501 L 16.1 57.801 L 35 57.801 A 31.555 31.555 0 0 0 38.351 57.635 Q 42.03 57.24 44.307 55.911 A 8.891 8.891 0 0 0 44.65 55.701 A 6.841 6.841 0 0 0 47.795 50.708 A 10.397 10.397 0 0 0 47.9 49.201 A 8.08 8.08 0 0 0 46.71 44.719 Q 43.982 40.501 35 40.501 Z M 32.2 12.201 L 16.1 12.201 L 16.1 28.701 L 32.2 28.701 Q 38.2 28.701 41.3 26.601 A 6.765 6.765 0 0 0 44.32 21.663 A 9.688 9.688 0 0 0 44.4 20.401 Q 44.4 16.301 41.3 14.251 A 10.492 10.492 0 0 0 38.567 13.001 Q 36.032 12.226 32.433 12.201 A 34.628 34.628 0 0 0 32.2 12.201 Z M 86.9 38.401 L 113.4 38.401 A 14.32 14.32 0 0 0 112.002 34.005 A 12.281 12.281 0 0 0 109 30.251 A 12.805 12.805 0 0 0 101.631 27.261 A 16.655 16.655 0 0 0 100.2 27.201 A 15.477 15.477 0 0 0 95.817 27.796 A 12.478 12.478 0 0 0 91.3 30.201 A 12.29 12.29 0 0 0 87.278 36.603 A 16.378 16.378 0 0 0 86.9 38.401 Z M 214 53.001 L 214 47.501 L 203.3 47.501 A 21.904 21.904 0 0 0 200.037 47.722 Q 195.838 48.358 194.422 50.82 A 5.876 5.876 0 0 0 193.7 53.801 A 6.151 6.151 0 0 0 194.118 56.109 Q 194.691 57.539 196.05 58.551 A 8.108 8.108 0 0 0 198.638 59.807 Q 199.769 60.139 201.124 60.247 A 17.186 17.186 0 0 0 202.5 60.301 Q 206.5 60.301 209.6 58.451 Q 212.7 56.601 214 53.001 Z"
          />
        </svg>
      </div>
    </div>
  </div>
  <h3
    class="pg-confirm__title"
  >
    Get Verified your email
    <svg
      class="pg-confirm__title__icon"
      fill="none"
      height="10"
      viewBox="0 0 10 10"
      width="10"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M3.33033 4.91061L0.367188 1.69265L1.83845 0.337891L4.68971 3.43434L7.54097 0.337891L9.01223 1.69265L6.04909 4.91061L9.0101 8.12624L7.53883 9.48101L4.68971 6.38688L1.84059 9.48101L0.369322 8.12624L3.33033 4.91061Z"
        fill="var(--system-red)"
        fill-rule="evenodd"
      />
    </svg>
  </h3>
  <div
    class="pg-confirm__content"
  >
    Something went wrong
  </div>
</div>
`;
